### PR TITLE
General maintenance to make the examples run again.

### PR DESCRIPTION
**Pull request by somethingelseentirely**
Hey,

I wanted to use these bindings for a project of mine, so I figured I'd polish them up a bit.

I've made changed to have them run on macOS, fixed some typos, and tried to improve the error handling and reporting situation a bit.

Thoughts and suggestions more than welcome! I'm fairly new to Zig :)

Cheers!

### CODE REVIEW

File: examples/memory.zig [hunk 1]
```diff
@@ -72,8 +79,8 @@ pub fn main() !void {
     };
     defer memory.deinit();
 
-    const pages = memory.pages();
-    const data_size = memory.size();
+    //const pages = memory.pages();
+    //const data_size = memory.size();
```
Comment (Luukdegram) [inline examples/memory.zig]: If they are unused, may as well remove them completely instead of commenting them out :)

File: examples/memory.zig [hunk 2]
```diff
@@ -17,18 +17,25 @@ const wat =
...
 pub fn main() !void {
-    var wat_bytes = wasmer.ByteVec.fromSlice(wat);
-    defer wat_bytes.deinit();
+    run () catch |err| {
+        const err_msg = try wasmer.lastError(std.heap.c_allocator);
+        defer std.heap.c_allocator.free(err_msg);
 
-    var wasm_bytes: wasmer.ByteVec = undefined;
-    wasmer.wat2wasm(&wat_bytes, &wasm_bytes);
+        std.debug.print("ERROR: {s}", .{err_msg});
```
Comment (Luukdegram) [inline examples/memory.zig]: As we are using `std.log` everywhere else, I think `std.log.err` would fit better here.

**Review by Luukdegram — CHANGES_REQUESTED** [review]
Thank you for taking the time to update this. It hasn't had any love in a while.
Left 2 nitpicks, other than that it looks good :)

File: examples/memory.zig (hunk 1 above)
Comment (somethingelseentirely) [reply examples/memory.zig]: I figured that they are there for educational purposes, and were simply ignored before :D
I can remove them though.

Comment (somethingelseentirely) [reply examples/memory.zig]: done!

File: examples/memory.zig (hunk 2 above)
Comment (somethingelseentirely) [reply examples/memory.zig]: done!

**Comment by Luukdegram** [comment]
Thanks!

**Pull request merged**